main.c

```diff
@@ -0,0 +1,438 @@
+/*
```

Comment:
Is it worth splitting the UI code and the "reading stuff out of streams data" code into separate files?

I do really wish there was a better alternative to doing all this in C.

Reply:
Actually is it worth splitting the UI code and "reading stuff out of streams data" into separate _programs_? Because then it might be possible to write sensible tests for the latter one (or it can just be unit tests of the C code I guess -- but tests of that would be nice)

Reply:
As a separate program I would have to pass data around, I think it can be effectively unit tested just split out to a different file.  Willing to keep the distinct program option on the table if we need it.